test/wait/awaitility.go

```diff
@@ -82,26 +93,74 @@ func UntilHasStatusCondition(conditions ...toolchainv1alpha1.Condition) CheInsta
 	}
 }
 
+// UntilHasTektonStatusCondition checks if TektonInstallation status has the given set of conditions
+func UntilHasTektonStatusCondition(conditions ...toolchainv1alpha1.Condition) TektonInstallationWaitCondition {
+	return func(a *ToolchainAwaitility, ic *v1alpha1.TektonInstallation) bool {
+		toolchain.AssertConditionsMatch(a.T, ic.Status.Conditions, conditions...)
+		if toolchain.ConditionsMatch(ic.Status.Conditions, conditions...) {
+			a.T.Logf("status conditions match in TektonInstallation '%s`", ic.Name)
+			return true
+		}
+		a.T.Logf("waiting for correct status condition of TektonInstallation '%s`", ic.Name)
+		return false
+	}
+}
+
 // WaitForCheInstallConditions waits until there is CheInstallation available with the given name and meeting the set of given wait-conditions
 func (a *ToolchainAwaitility) WaitForCheInstallConditions(name string, waitCond ...CheInstallationWaitCondition) error {
 	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
-		ic := &v1alpha1.CheInstallation{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, ic); err != nil {
+		ci := &v1alpha1.CheInstallation{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Name: name}, ci); err != nil {
 			if errors.IsNotFound(err) {
 				a.T.Logf("waiting for availability of CheInstallation '%s'", name)
 				return false, nil
 			}
 			return false, err
 		}
 		for _, isMatched := range waitCond {
-			if !isMatched(a, ic) {
+			if !isMatched(a, ci) {
 				return false, nil
 			}
 		}
 		return true, nil
 	})
 }
 
+// WaitForTektonInstallConditions waits until there is TektonInstallation available with the given name and meeting the set of given wait-conditions
+func (a *ToolchainAwaitility) WaitForTektonInstallConditions(name string, waitCond ...TektonInstallationWaitCondition) error {
+	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		ti := &v1alpha1.TektonInstallation{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Name: name}, ti); err != nil {
+			if errors.IsNotFound(err) {
+				a.T.Logf("waiting for availability of TektonInstallation '%s'", name)
+				return false, nil
+			}
+			return false, err
+		}
+		for _, isMatched := range waitCond {
+			if !isMatched(a, ti) {
+				return false, nil
+			}
+		}
```

Comment:
see if you can switch this way of comparing conditions https://github.com/codeready-toolchain/toolchain-e2e/blob/master/wait/member.go#L63-L66

Reply:
not sure how would it help?

Reply:
Just a diff_way and I found it simple so suggested but anyway fine with the way it's done currently :+1: